DuplicateAssemblyScanner/DuplicateAssemblyScanner/Settings.cs

```diff
@@ -0,0 +1,124 @@
+namespace DuplicateAssemblyScanner {
+    using ColossalFramework.UI;
+    using DuplicateAssemblyScanner.Structs;
+    using DuplicateAssemblyScanner.Util;
+    using ICities;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Generates the settings screen based on cached results from the assembly scanner.
+    /// </summary>
+    public class Settings {
+        // Markers used in checkbox labels
+        private const string MARKER_BLANK = " ";
+        private const string MARKER_ENABLED = "*";
+
+        // Cache of scan results
+        private static Dictionary<string, int> duplicates_;
+        private static Dictionary<string, List<ModAssembly>> occurrences_;
+        private static bool duplicatesFound_;
+
+        /// <summary>
+        /// Generate the options screen listing all the duplicates (if found).
+        /// </summary>
+        /// 
+        /// <param name="helper">UI helper for building settings screen.</param>
+        public static void CreateUI(UIHelperBase helper) {
+
+            UIHelperBase group;
+
+            if (CacheScan(
+                out Dictionary<string, int> duplicates,
+                out Dictionary<string, List<ModAssembly>> occurrences)) {
+
+                group = helper.AddGroup("Duplicate assemblies were detected.");
+
+                Disable((UICheckBox)group.AddCheckbox("= Assembly is loaded in to app domain", true, NoOp));
+                Disable((UICheckBox)group.AddCheckbox("= Assembly not loaded", false, NoOp));
+
+                foreach (KeyValuePair<string, int> duplicate in duplicates) {
+
+                    if (duplicate.Value > 1) {
+
+                        if (occurrences.TryGetValue(duplicate.Key, out List<ModAssembly> mods)) {
+
+                            RenderDetails(helper, duplicate.Key, duplicate.Value, mods);
+                        }
+                    }
+                }
+            } else {
+                helper.AddGroup("No duplicates.");
+            }
+        }
+
+        /// <summary>
+        /// A dummy click handler for checkboxes. Does nothing.
+        /// </summary>
+        /// 
+        /// <param name="_">Ignored paramter.</param>
+        internal static void NoOp(bool _) {
+            // do nothing
+        }
+
+        /// <summary>
+        /// Renders UI for a given assembly and the mods that contain it.
+        /// </summary>
+        /// 
+        /// <param name="helper">Helper for creating UI.</param>
+        /// <param name="assemblyName">The name of the assembly.</param>
+        /// <param name="loaded">How many copies of the assembly are loaded.</param>
+        /// <param name="list">The list of mods containing the assembly.</param>
+        internal static void RenderDetails(
+            UIHelperBase helper,
+            string assemblyName,
+            int loaded,
+            List<ModAssembly> list) {
```

Comment:
`list` could be `IReadOnlyCollection<ModAssembly>`

Reply:
Updated but curious to know what the benefit is?

Reply:
Ah, didn't work - we're on .Net 3.5 and the package required for `IReadOnlyCollection` requires later version. :(

Reply:
Sorry, I always forget we're stuck in the Stone Age with Unity 5...